src/sumlog.jl

```diff
@@ -1,42 +1,94 @@
 """
-$(SIGNATURES)
+    sumlog(X::AbstractArray{T}; dims)
```

Comment:
```suggestion
    sumlog(x::AbstractArray{<:Real}; dims=:)
```

Reply:
I agree with `x` instead of `X`. I tried to match the original docs, but now I see they use lowercase. I may have been looking at an old version.

I think in general it makes sense to match the Julia docs for `sum`. For arrays, that looks like `sum(f, A::AbstractArray; dims)`.

We've already discussed that we don't need `<:Real` for dispatch, and it disallows some types that would be useful to include.